src/uk/zenko/src/eu/kanade/tachiyomi/extension/uk/zenko/Zenko.kt

```diff
@@ -0,0 +1,193 @@
+        val url = "$API_URL/titles/$mangaId/chapters"
+        return GET(url, headers)
+    }
+
+    override fun chapterListParse(response: Response): List<SChapter> {
+        val result = response.parseAs<List<ChapterResponseItem>>()
+        return result.sortedByDescending { item ->
+            val id = StringProcessor.generateId(item.name)
+            if (id > 0) id else item.id.toDouble()
+        }.map { chapterResponseItem ->
+            SChapter.create().apply {
+                url = "/titles/${chapterResponseItem.titleId}/${chapterResponseItem.id}"
+                name = StringProcessor.format(chapterResponseItem.name)
+                date_upload = chapterResponseItem.createdAt!!.secToMs()
+                scanlator = chapterResponseItem.publisher!!.name
+            }
+        }
+    }
+
+    // =============================== Pages ================================
+    override fun pageListRequest(chapter: SChapter): Request {
+        val chapterId = chapter.url.substringAfterLast('/')
+        val url = "$API_URL/chapters/$chapterId"
+        return GET(url, headers)
+    }
+
+    override fun pageListParse(response: Response): List<Page> {
+        val data = response.parseAs<ChapterResponseItem>()
+        return data.pages!!.map { page ->
+            Page(page.id, imageUrl = "$IMAGE_STORAGE_URL/${page.imgUrl}")
+        }
+    }
+
+    override fun imageUrlParse(response: Response): String = ""
+
+    // ============================= Utilities ==============================
+    private fun parseAsMangaResponseDto(response: Response): MangasPage {
+        val zenkoMangaListResponse = response.parseAs<ZenkoMangaListResponse>()
+        return makeMangasPage(zenkoMangaListResponse.data, zenkoMangaListResponse.meta.hasNextPage)
+    }
+
+    private fun offsetCounter(page: Int): Int {
+        return if (page == 1) {
+            0 // for page 1 offset must be 0
+        } else {
+            (page - 1) * 15 // calculate offset for other pages
+        }
+    }
+
+    private fun makeZenkoMangaRequest(offset: Int, sortBy: String): Request {
+        val url = "$API_URL/titles".toHttpUrl().newBuilder()
+            .addQueryParameter("limit", "15")
+            .addQueryParameter("offset", "$offset")
+            .addQueryParameter("sortBy", sortBy)
+            .addQueryParameter("order", "DESC")
+            .build()
+        return GET(url, headers)
+    }
+
+    private fun makeMangasPage(
+        mangaList: List<MangaDetailsResponse>,
+        hasNextPage: Boolean = false,
+    ): MangasPage {
+        return MangasPage(
+            mangaList.map(::makeSManga),
+            hasNextPage,
+        )
+    }
+
+    private fun makeSManga(mangaDto: MangaDetailsResponse) = SManga.create().apply {
+        title = mangaDto.engName ?: mangaDto.name
+        thumbnail_url =
+            "$IMAGE_STORAGE_URL/${mangaDto.coverImg}?optimizer=image&width=560&quality=70&height=auto"
+        url = "/titles/${mangaDto.id}"
```

Comment:
Be careful with building URLs like this, especially in regards to encoding. Consider using `HttpUrlBuilder`